probe-rs/src/debug/variable.rs

```diff
@@ -593,28 +632,33 @@ impl Variable {
     pub fn extract_value(
         &mut self,
         memory: &mut dyn MemoryInterface,
-        variable_cache: &variable_cache::VariableCache,
+        variable_cache: &VariableCache,
     ) {
         if let VariableValue::Error(_) = self.value {
             // Nothing more to do ...
             return;
         }
 
-        if !self.value.is_empty()
-        // The value was set explicitly, so just leave it as is, or it was an error, so don't attempt anything else
-        || !self.memory_location.valid()
-        // This may just be that we are early on in the process of `Variable` evaluation
-        || matches!(self.type_name.inner(), VariableType::Unknown)
+        if !self.value.is_empty() {
```

Comment:
Should we maybe write these 3 conditions similar to how `update_value` has it in L557-560? Right now you refactored the same pattern in two different ways.